### ISSUE TITLE
[Monitoring V1] v0.2.0 should not show on 1.15 cluster 

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#30453]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
- use rke v1.1.9 to create a k8s `v1.15.12-rancher2-6` cluster 
- import the cluster into a rancher:`v2.5.3` setup 
![image](https://user-images.githubusercontent.com/6218999/101963235-7870f280-3bcb-11eb-9ada-db8dc3cda0b1.png)

- edit the branch of the `system-library` to `dev-v2.5`
- go to the cluster monitoring page of the cluster 

**Result:**
- `0.2.0` shows in the version list
![image](https://user-images.githubusercontent.com/6218999/101963277-8cb4ef80-3bcb-11eb-9e4a-4ab01670a1fb.png)

Expected:
- `0.2.0` should not show in the list, it should show up only when the k8s version is 1.16 or higher.

**Comment by aiyengar2** [comment]
@rmweir could you look into this since it's related to changes from https://github.com/rancher/rancher/pull/29559? The [kubeVersion in the monitoring chart](https://github.com/rancher/system-charts/blob/aefdee237acd5c89ed7e6225f9040f51707f562a/charts/rancher-monitoring/v0.2.0/Chart.yaml#L12) seems to be correct based on the fact that it matches the [kubeVersion in upstream](https://github.com/prometheus-community/helm-charts/blob/f48f309cfda084b16beeed50646b5e3aae61a6bf/charts/kube-prometheus-stack/Chart.yaml#L23).

If the kubeVersion needs to be modified, I can make the fix but this seems like an edge case that should be handled by catalog v1 for charts that have similarly structured kubeVersions as `kube-prometheus-stack`.

**Comment by maggieliu** [comment]
@jiaqiluo Please test with a 2.5.4 setup because the UI filtering is added in 2.5.4 [here](https://github.com/rancher/rancher/issues/29774)

**Comment by jiaqiluo** [comment]
@maggieliu This is a really good point! 

I passed a valid version when installing Rancher:
```
sudo docker run -d --privileged -e CATTLE_SERVER_VERSION=2.5.4  --restart=unless-stopped -p 80:80 -p 443:443 rancher/rancher:v2.5.4-rc4
```

The good news is that monitoring `0.2.0` does NOT show on the k8s 1.15 cluster( only 0.1.4 shows in the list).

But there are both `0.1.4` and `0.2.0` showing on a k8s 1.16 cluster.
We expect that only the latest/highest version shows, so in this case, it should be only `0.2.0`. 


![image](https://user-images.githubusercontent.com/6218999/102127014-261d1500-3e09-11eb-9930-2198360888da.png)

**Comment by maggieliu** [comment]
Will add the min Rancher version to 2.5.4. It is fine to have both versions showing because one of them is for k8s 1.15 in case user hasn't upgraded their k8s.

**Comment by jiaqiluo** [comment]
The fix is validated on rancher:v2.5.3 

Steps:
- run rancher:v2.5.3 single install
- edit the system-library chart to points to the branch `dev-v2.5`
- go to the cluster monitoring page 

Result:
- only `0.1.4` shows in the version list 

<img width="764" alt="Screen Shot 2020-12-16 at 10 44 04 AM" src="https://user-images.githubusercontent.com/6218999/102386430-47f2d500-3f8c-11eb-935f-c9c6fd9a5e60.png">
